src/Sentry/Internal/GarbageCollectionMonitor.cs

```diff
@@ -10,8 +11,13 @@ internal sealed class GarbageCollectionMonitor
     private const int MaxGenerationThreshold = 10;
     private const int LargeObjectHeapThreshold = 10;
 
-    public static Task Start(Action onGarbageCollected, CancellationToken cancellationToken, IGCImplementation? gc = null) =>
-        Task.Run(() => MonitorGarbageCollection(onGarbageCollected, cancellationToken, gc), cancellationToken);
+    public static Task Start(Action onGarbageCollected, CancellationToken cancellationToken, IGCImplementation? gc = null)
+    {
+        using (new SuppressBlockingDetection())
```

Comment:
@jamescrosswell this is the blocking detector itself right? we shouldn't add this then?

Reply:
Yeah don't do that...